File: code_samples/java/DesignPatternsObserver.java
/**
 * Design Patterns: Observer
 * AI/ML Training Sample
 */
public class DesignPatternsObserver {
    
    private String data;
    
    public DesignPatternsObserver() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        DesignPatternsObserver instance = new DesignPatternsObserver();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}

File: code_samples/go/design_patterns_observer.go
package main

import (
    "fmt"
)

// Design Patterns: Observer
// AI/ML Training Sample

type Observer struct {
    Data string
}

func NewObserver() *Observer {
    return &Observer{
        Data: "",
    }
}

func (s *Observer) Process(input string) {
    s.Data = input
}

func (s *Observer) Validate() bool {
    return len(s.Data) > 0
}

func (s *Observer) GetData() string {
    return s.Data
}

func main() {
    instance := NewObserver()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
